netty/src/main/java/io/grpc/netty/NettyChannelBuilder.java

```diff
@@ -637,14 +635,16 @@ public int getDefaultPort() {
 
     private boolean closed;
 
-    NettyTransportFactory(ProtocolNegotiator protocolNegotiator,
+    NettyTransportFactory(
+        ProtocolNegotiator.ClientFactory protocolNegotiator,
         ChannelFactory<? extends Channel> channelFactory,
         Map<ChannelOption<?>, ?> channelOptions, ObjectPool<? extends EventLoopGroup> groupPool,
         boolean autoFlowControl, int flowControlWindow, int maxMessageSize, int maxHeaderListSize,
         long keepAliveTimeNanos, long keepAliveTimeoutNanos, boolean keepAliveWithoutCalls,
         TransportTracer.Factory transportTracerFactory, LocalSocketPicker localSocketPicker,
         boolean useGetForSafeMethods) {
-      this.protocolNegotiator = checkNotNull(protocolNegotiator, "protocolNegotiator");
+      this.protocolNegotiator =
+          checkNotNull(protocolNegotiator, "protocolNegotiator").newNegotiator();
```

Comment:
Checking for null for `protocolNegotiator` does nothing.

```suggestion
          checkNotNull(protocolNegotiator.newNegotiator(), "protocolNegotiator instance");
```

Reply:
It was a trivial check for the input, because the constructor is called by other classes, the input is out of control of this class itself. Checking null for the return value of `newNegotiator()` seems not necessary - we can assume the return value is not null. Removed `checkNotNull()` entirely.